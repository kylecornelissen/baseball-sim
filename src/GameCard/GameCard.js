import React from 'react';
import { Link } from 'react-router-dom';

import './GameCard.scss';

export const GameCard = ({game}) => {
  return (
    <Link to={`/game/${game.id}`}>
      <h3>{game.awayTeam.name} @ {game.homeTeam.name}</h3>
    </Link>
  )
}

export default GameCard;
