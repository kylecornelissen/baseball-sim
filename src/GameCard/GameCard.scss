@import '../variables.scss';

.game-card-link {
  color: $blueaqua;
  text-decoration: none;
  margin: 15px;
}

.game-card {
  border: 3px solid $blueaqua;
  border-radius: 4px;
  color: $white;
  font-size: 22px;
  margin: 0;
  padding: 0%;
  width: 350px;
}

.team-logo {
  object-fit: contain;
  height: 35px;
  width: 35px;
  padding: 5px;
  background: lightBlue;
  box-shadow: inset 0px 0px 8px 2px #04c6fb;
}

.away-team-container, .home-team-container {
  border-bottom: 1px solid $blueaqua;
  border-top: 1px solid $blueaqua;
  margin: 0;
  margin: 0;
  padding: 0%;
  display: grid;
  grid-template-columns: 15% 70% 15%;
  grid-template-areas:
    "logo name score";
  text-align: left;
  >h3 {
    font-size: 22px;
    font-weight: 400;
    text-align: left;
    margin: 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
  >span {
    grid-area: score;
    border-left: 2px solid $blueaqua;
    font-weight: 700;
    margin-right: 4%;
    text-align: center;
    width: 49px;
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
}

.game-card:hover {
  border: 3px solid $white;
  box-shadow: inset 0 0 20px 6px $blueaqua;
  transition: .3s ease;
  >.away-team-container, .home-team-container {
    border-bottom: 1px solid $white;
    border-top: 1px solid $white;
    >img {
      // background: $blueaqua;
    }
    >span {
      border-left: 2px solid $white;
    }
  }
}
