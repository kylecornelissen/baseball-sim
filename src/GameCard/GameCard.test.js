import React from 'react';
import { shallow } from 'enzyme';
import GameCard from './GameCard';

describe('GameCard', () => {
  const game = {awayTeam: {name: "angels"}, homeTeam: {name: "athletics"}};

  it('renders the names of the teams in <h3> tags', () => {
    const wrapper = shallow(<GameCard game={game} />);
    const title1 = <h3>angels</h3>;
    const title2 = <h3>athletics</h3>;

    expect(wrapper.contains(title1)).toEqual(true);
    expect(wrapper.contains(title2)).toEqual(true);
  });

  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<GameCard
      game={game}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  
});
