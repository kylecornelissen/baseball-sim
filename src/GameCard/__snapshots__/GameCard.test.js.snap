// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCard should match the snapshot with all data passed in correctly 1`] = `
<Link
  className="game-card-link"
  to={
    Object {
      "pathname": "/game/undefined",
      "state": Object {
        "awayTeam": Object {
          "name": "angels",
        },
        "homeTeam": Object {
          "name": "athletics",
        },
      },
    }
  }
>
  <article
    className="game-card"
  >
    <div
      className="away-team-container"
    >
      <img
        alt="angels logo"
        className="team-logo"
        src="/mlb-logos/angels.svg"
      />
      <h3>
        angels
      </h3>
      <span />
    </div>
    <div
      className="home-team-container"
    >
      <img
        alt="athletics logo"
        className="team-logo"
        src="/mlb-logos/athletics.svg"
      />
      <h3>
        athletics
      </h3>
      <span />
    </div>
  </article>
</Link>
`;
